diff --git a/vboxdrv/r0drv/linux/timer-r0drv-linux.c b/vboxdrv/r0drv/linux/timer-r0drv-linux.c
index 79655dc..d6c50ea 100644
--- a/vboxdrv/r0drv/linux/timer-r0drv-linux.c
+++ b/vboxdrv/r0drv/linux/timer-r0drv-linux.c
@@ -50,7 +50,7 @@
 # define RTTIMER_LINUX_WITH_HRTIMER
 #endif
 
-#if LINUX_VERSION_CODE < KERNEL_VERSION(2, 6, 31)
+#if LINUX_VERSION_CODE < KERNEL_VERSION(2, 6, 31) || LINUX_VERSION_CODE >= KERNEL_VERSION(4, 8, 0)
 # define mod_timer_pinned               mod_timer
 # define HRTIMER_MODE_ABS_PINNED        HRTIMER_MODE_ABS
 #endif
