diff -uprNbd a/vboxnetadp/linux/VBoxNetAdp-linux.c b/vboxnetadp/linux/VBoxNetAdp-linux.c
--- a/vboxnetadp/linux/VBoxNetAdp-linux.c	2016-04-20 13:29:13.000000000 +0300
+++ b/vboxnetadp/linux/VBoxNetAdp-linux.c	2016-05-30 07:05:52.915567154 +0300
@@ -147,13 +147,14 @@ static int vboxNetAdpLinuxStop(struct ne
 
 static int vboxNetAdpLinuxXmit(struct sk_buff *pSkb, struct net_device *pNetDev)
 {
+    struct netdev_queue *txq = netdev_get_tx_queue(pNetDev, skb_get_queue_mapping(pSkb));
     PVBOXNETADPPRIV pPriv = netdev_priv(pNetDev);
 
     /* Update the stats. */
     pPriv->Stats.tx_packets++;
     pPriv->Stats.tx_bytes += pSkb->len;
     /* Update transmission time stamp. */
-    pNetDev->trans_start = jiffies;
+    txq->trans_start = jiffies;
     /* Nothing else to do, just free the sk_buff. */
     dev_kfree_skb(pSkb);
     return 0;
