diff -rua asterisk-13.9.1.old/configure.ac asterisk-13.9.1.new/configure.ac
--- asterisk-13.9.1.old/configure.ac	2016-06-23 12:33:15.164525883 +0200
+++ asterisk-13.9.1.new/configure.ac	2016-06-23 12:38:25.806511388 +0200
@@ -2168,6 +2168,7 @@
 
 if test "$USE_PJPROJECT" != "no" ; then
    if test "$PJPROJECT_BUNDLED" = "yes" ; then
+       AC_CONFIG_MACRO_DIR(third-party/pjproject)
        PJPROJECT_CONFIGURE([$PJPROJECT_DIR])
    else
       AST_PKG_CONFIG_CHECK([PJPROJECT], [libpjproject])
